internal/tsoptions/wildcarddirectories.go

```diff
@@ -110,21 +95,55 @@ type wildcardDirectoryMatch struct {
 }
 
 func getWildcardDirectoryFromSpec(spec string, useCaseSensitiveFileNames bool) *wildcardDirectoryMatch {
-	match, _ := wildcardDirectoryPattern.FindStringMatch(spec)
-	if match != nil {
-		// We check this with a few `Index` calls because it's more efficient than complex regex
-		questionWildcardIndex := strings.Index(spec, "?")
-		starWildcardIndex := strings.Index(spec, "*")
-		lastDirectorySeparatorIndex := strings.LastIndexByte(spec, tspath.DirectorySeparator)
-
-		// Determine if this should be watched recursively
-		recursive := (questionWildcardIndex != -1 && questionWildcardIndex < lastDirectorySeparatorIndex) ||
-			(starWildcardIndex != -1 && starWildcardIndex < lastDirectorySeparatorIndex)
-
-		return &wildcardDirectoryMatch{
-			Key:       toCanonicalKey(match.String(), useCaseSensitiveFileNames),
-			Path:      match.String(),
-			Recursive: recursive,
+	// Find the first occurrence of wildcards (* or ?)
+	questionWildcardIndex := strings.Index(spec, "?")
+	starWildcardIndex := strings.Index(spec, "*")
+
+	// Find the earliest wildcard
+	firstWildcardIndex := -1
+	if questionWildcardIndex != -1 && starWildcardIndex != -1 {
+		if questionWildcardIndex < starWildcardIndex {
+			firstWildcardIndex = questionWildcardIndex
+		} else {
+			firstWildcardIndex = starWildcardIndex
+		}
+	} else if questionWildcardIndex != -1 {
+		firstWildcardIndex = questionWildcardIndex
+	} else if starWildcardIndex != -1 {
+		firstWildcardIndex = starWildcardIndex
+	}
```

Comment:
Consider

```suggestion
	// Find the earliest wildcard
	firstWildcardIndex := strings.IndexAny(spec, "?*")
	
	// Find the first occurrence of wildcards (* or ?)
	starWildcardIndex := -1
	questionWildcardIndex := -1
	if firstWildcardIndex != -1 {
		if spec[firstWildcardIndex] == '?' {
			questionWildcardIndex = firstWildcardIndex
			starWildcardIndex = core.IndexAfter(spec, "*")
		} else {
			firstWildcardIndex = starWildcardIndex
			questionWildcardIndex = core.IndexAfter(spec, "?")
		}
	}
```

though I wonder if that ends up being slower than just a naive run through the string for each character.

Another way to simplify might be

```suggestion
	// Find the first occurrence of wildcards (* or ?)
	questionWildcardIndex := strings.IndexByte(spec, '?')
	starWildcardIndex := strings.IndexByte(spec, '*')

	// Find the earliest wildcard
	firstWildcardIndex := -1
	if questionWildcardIndex == -1 || starWildcardIndex == -1 {
		firstWildcardIndex = max(questionWildcardIndex, starWildcardIndex)
	} else {
		firstWildcardIndex = min(questionWildcardIndex, starWildcardIndex)
	}
```